include/dlaf/communication/kernels/broadcast.h

```diff
@@ -58,67 +59,88 @@ void recvBcast(const matrix::Tile<T, D>& tile, comm::IndexT_MPI root_rank,
+      internal::transformMPI(sendBcast_o) | start_detached();
 }
 
 namespace internal {
 
 template <class T>
 struct ScheduleRecvBcast {
   template <Device D>
-  static auto call(const comm::Executor& ex, pika::future<matrix::Tile<T, D>> tile,
-                   comm::IndexT_MPI root_rank, pika::future<common::PromiseGuard<Communicator>> pcomm) {
+  static auto call(pika::future<matrix::Tile<T, D>> tile, comm::IndexT_MPI root_rank,
+                   pika::future<common::PromiseGuard<Communicator>> pcomm) {
 #if !defined(DLAF_WITH_CUDA_RDMA)
     static_assert(D == Device::CPU, "With CUDA RDMA disabled, MPI accepts just CPU memory.");
 #endif
 
-    using pika::dataflow;
-    using matrix::unwrapExtendTiles;
+    using dlaf::internal::whenAllLift;
+    using pika::execution::experimental::start_detached;
 
-    return dataflow(ex, unwrapExtendTiles(recvBcast_o), std::move(tile), root_rank, std::move(pcomm));
+    return whenAllLift(std::move(tile), root_rank, std::move(pcomm)) | transformMPI(recvBcast_o);
   }
 
 #if defined(DLAF_WITH_CUDA) && !defined(DLAF_WITH_CUDA_RDMA)
-  static void call(const comm::Executor& ex, pika::future<matrix::Tile<T, Device::GPU>> tile,
-                   comm::IndexT_MPI root_rank, pika::future<common::PromiseGuard<Communicator>> pcomm) {
-    using matrix::duplicateIfNeeded;
-
+  static auto call(pika::future<matrix::Tile<T, Device::GPU>> tile_gpu, comm::IndexT_MPI root_rank,
+                   pika::future<common::PromiseGuard<Communicator>> pcomm) {
     // Note:
-    //
-    // TILE_GPU -+-> duplicateIfNeeded<CPU> ---> TILE_CPU ---> recvBcast ---> TILE_CPU -+-> copy
-    //           |                                                                      |
-    //           +----------------------------------------------------------------------+
-    //
-    // Actually `duplicateIfNeeded` always makes a copy, because it is always needed since this
-    // is the specialization for GPU input and MPI without CUDA_RDMA requires CPU memory.
-
-    auto tile_gpu = tile.share();
-    auto tile_cpu = duplicateIfNeeded<Device::CPU>(tile_gpu);
-
-    tile_cpu = std::move(pika::get<0>(pika::split_future(
-        ScheduleRecvBcast<T>::call(ex, std::move(tile_cpu), root_rank, std::move(pcomm)))));
-
-    pika::execution::experimental::when_all(dlaf::internal::keepIfSharedFuture(std::move(tile_cpu)),
-                                            dlaf::internal::keepIfSharedFuture(std::move(tile_gpu))) |
-        dlaf::matrix::copy(
-            dlaf::internal::Policy<dlaf::matrix::internal::CopyBackend_v<Device::CPU, Device::GPU>>()) |
-        pika::execution::experimental::start_detached();
+    // TILE_GPU -+-> duplicate<CPU> ---> TILE_CPU ---> recvBcast ---> TILE_CPU -+-> copy
+    //           |                                                              |
+    //           +--------------------------------------------------------------+
```

Comment:
This diagram doesn't look correct, but it was already before the changes...
Shouldn't it be  
```
TILE_GPU -+-> duplicate<CPU> ---> TILE_CPU ---> recvBcast ---> TILE_CPU -+-> copy -+->
          +-----------------------------------> recvBcast -------------------------+ 
```
?

Reply:
Hmm, in your diagram the GPU tile is passed into `recvBcast` together with the copied CPU tile? Or what is the difference you're thinking of? I think the original diagram is accurate in that the GPU tile is first duplicated, and then used again after the `recvBcast` is done to be filled with the received data.

Reply:
Ok I understand it now...
I misinterpreted the diagram... thinking it represented the two possible paths with and without direct GPU comm...